dhall/tests/Dhall/Test/QuickCheck.hs

```diff
@@ -397,6 +410,10 @@ tests =
           , Test.QuickCheck.property isSameAsSelf
           , QuickCheckTests 10000
           )
+        , ( "Inferred types should be normalized"
```

Comment:
Note that this is currently not one of the four soundness properties that the language aims to enforce:

https://github.com/dhall-lang/dhall-lang/blob/5fea4544f6fe57acc034828a0a6ccebfd52aad15/.github/CONTRIBUTING.md#soundness-of-proposed-changes

Specifically, the language currently only guarantees that inferred types are safe to normalize (i.e. the inferred type does not itself need to type-checked again before being normalized).  We could add the guarantee that you describe, but that guarantee should be upstreamed into the standard.

Reply:
Also, such a guarantee would imply other changes to the standard (i.e. we would have to replace all occurrences of `:→` with `:`, for example)

Reply:
Thanks for the pointer to the soundness properties! I have opened https://github.com/dhall-lang/dhall-lang/issues/724 to discuss the potential standard change.